crates/sc-proof-of-time/src/lib.rs

```diff
@@ -0,0 +1,86 @@
+//! Subspace proof of time implementation.
+
+mod clock_master;
+mod pot_client;
+mod pot_state;
+mod utils;
+
+use crate::utils::GOSSIP_PROTOCOL;
+use sc_network::config::NonDefaultSetConfig;
+use subspace_core_primitives::{BlockHash, BlockNumber, PotKey, SlotNumber};
+
+pub use clock_master::ClockMaster;
+pub use pot_client::PotClient;
+
+#[derive(Debug, Clone)]
+pub struct PotConfig {
+    /// Frequency of entropy injection from consensus.
+    pub randomness_update_interval_blocks: BlockNumber,
+
+    /// Starting point for entropy injection from consensus.
+    pub injection_depth_blocks: BlockNumber,
+
+    /// Number of slots it takes for updated global randomness to
+    /// take effect.
+    pub global_randomness_reveal_lag_slots: SlotNumber,
+
+    /// Number of slots it takes for injected randomness to
+    /// take effect.
+    pub pot_injection_lag_slots: SlotNumber,
+
+    /// If the received proof is more than max_future_slots into the
+    /// future from the current tip's slot, reject it.
+    pub max_future_slots: SlotNumber,
+
+    /// Number of checkpoints per proof.
+    pub num_checkpoints: u8,
+
+    /// Number of EAS iterations per checkpoints.
+    /// Total iterations per proof = num_checkpoints * checkpoint_iterations.
```

Comment:
I've received feedback that it is better to define the number of iterations the other way around. This `PotConfig` struct should specify `pot_iterations`, which would be the total iterations in one slot and `checkpoint_iterations` should be computed inside functions where it is used as `checkpoint_iterations = pot_iterations/num_checkpoints`. I'm sorry this has changed recently.